include/libp2p/crypto/common_functions.hpp

```diff
@@ -47,6 +49,28 @@ namespace libp2p::crypto {
   NewEvpPkeyFromBytes(int, gsl::span<const uint8_t>,
                       decltype(EVP_PKEY_new_raw_public_key) *);
 
+  /**
+   * @brief Generate EC signature based on key type
+   * @param digest - message hash
+   * @param key - private key
+   * @param output - signature result
+   * @return EC signature or error code
+   */
+  outcome::result<std::vector<uint8_t>> GenerateEcSignature(
+      gsl::span<const uint8_t> digest, const std::shared_ptr<EC_KEY> &key);
+
+  /**
+   * @brief Verify EC signature based on key type
+   * @param digest - message hash
+   * @param signature - bytes of the EC signature
+   * @param key - EC public key
+   * @return signature status or error code
+   */
+  outcome::result<bool> VerifyEcSignature(
+      const std::array<uint8_t, 32u> &digest,
+      const std::vector<uint8_t> &signature,
```

Comment:
WE can use gsl span instead

Reply:
Changed to gsl::span<const uint8_t> for digest and signature